backend/src/Designer/Services/Implementation/RepositorySI.cs

```diff
@@ -858,10 +859,77 @@ public List<ServiceResource> GetServiceResources(string org, string repository,
             return serviceResourceList;
         }
 
+        public ActionResult UpdateServiceResource(string org, string id, ServiceResource updatedResource)
+        {
+            if (updatedResource != null && id == updatedResource.Identifier)
+            {
+                string repository = string.Format("{0}-resources", org);
+                List<FileSystemObject> resourceFiles = GetResourceFiles(org, repository);
+                string repopath = _settings.GetServicePath(org, repository, AuthenticationHelper.GetDeveloperUserName(_httpContextAccessor.HttpContext));
+
+                foreach (FileSystemObject resourceFile in resourceFiles)
+                {
+                    string jsonString = File.ReadAllText($"{repopath}/{resourceFile.Path}");
+                    ServiceResource serviceResource = JsonConvert.DeserializeObject<ServiceResource>(jsonString);
+
+                    if (serviceResource != null && serviceResource.Identifier == updatedResource.Identifier)
+                    {
+                        string updatedResourceString = JsonConvert.SerializeObject(updatedResource);
+                        File.WriteAllText($"{repopath}/{resourceFile.Path}", updatedResourceString);
+                        return new StatusCodeResult(201);
+                    }
+                }
+            }
+            else
+            {
+                return new StatusCodeResult(400);
+            }
+
+            return new StatusCodeResult(403);
+        }
+
+        public ActionResult AddServiceResource(string org, ServiceResource newResource)
```

Comment:
How do we now that this serviceresource is a app resource or a generic resource?

Reply:
That is a good question, currently we don't. I must add some way to determine that upon creation.